src/Core/Managed/Shared/TelemetryClientExtensions.cs

```diff
@@ -93,26 +94,78 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                 operationTelemetry.GenerateOperationId();
             }
 
+            var telemetryContext = operationTelemetry.Context.Operation;
+
             // If the operation is not executing in the context of any other operation
             // set its name and id as a context (root) operation name and id
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Id))
+            if (string.IsNullOrEmpty(telemetryContext.Id))
             {
-                operationTelemetry.Context.Operation.Id = operationTelemetry.Id;
+                telemetryContext.Id = operationTelemetry.Id;
             }
 
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Name))
+            if (string.IsNullOrEmpty(telemetryContext.Name))
             {
-                operationTelemetry.Context.Operation.Name = operationTelemetry.Name;
+                telemetryContext.Name = operationTelemetry.Name;
             }
 
+            bool isActivityEnabled = false;
+#if !NET40
+            if (ActivityExtensions.IsActivityEnabled())
+            {
+                bool operationNameIsSet = false;
+
+                var childActivity = new Activity("Microsoft.ApplicationInsights.OperationContext");
+                if (!string.IsNullOrEmpty(telemetryContext.Name))
+                {
+                    childActivity.SetOperationName(telemetryContext.Name);
+                    operationNameIsSet = true;
+                }
+
+                var parentActivity = Activity.Current;
+
+                if (parentActivity == null)
+                {
+                    if (!string.IsNullOrEmpty(telemetryContext.Id))
+                    {
+                        childActivity.SetParentId(telemetryContext.Id);
+                    }
+                    else if (!string.IsNullOrEmpty(telemetryContext.ParentId))
+                    {
+                        childActivity.SetParentId(telemetryContext.ParentId);
+                    }
+
+                    operationTelemetry.Id = childActivity.ParentId;
```

Comment:
I agree that it is a breaking change.

We could consider generating default telemetry Id from the Activity.Current **before** initialization in OperatonTelemetry ctor, however Activity.Current may be null (because Activity is not started yet).